### PR TITLE
Adding parser

### PR DESCRIPTION
**Pull request by camertron**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+source 'https://rubygems.org'
+
+gemspec
+
+group :development, :test do
```
Comment (zvkemp) [inline Gemfile]: Recommend moving these to the gemspec (using `s.add_development_dependency`). This file should end at line 3 unless you need to specify the git location of a dependency.

File: Rakefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# encoding: UTF-8
+
+$:.unshift File.join(File.dirname(__FILE__), 'lib')
+
+require 'rubygems' unless ENV['NO_RUBYGEMS']
```
Comment (zvkemp) [inline Rakefile:5]: Is this for 1.8 compatibility?

File: lib/tmx-parser/elements.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+  class PropertyValue
+    attr_accessor :value
+
+    def initialize
+      @value = ''
+    end
+
+    def receive_text(str)
+      @value += str
```
Comment (zvkemp) [inline lib/tmx-parser/elements.rb]: Prefer `@value << str` for mutating strings in place (ditto for lines 52, 65, 78).

File: lib/tmx-parser/elements.rb [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    def initialize(i)
+      @i = i
+      @text = ''
+    end
+
+    def receive_text(str)
+      @text += str
+    end
+  end
+
```
Comment (zvkemp) [inline lib/tmx-parser/elements.rb:84]: Any reason `BeginPair` and `EndPair` need to be distinct?

File: lib/tmx-parser/listener.rb [hunk 5]
```diff
@@ -0,0 +1,80 @@
+# encoding: UTF-8
+
+module TmxParser
+
+  class Listener
+    include TagNames
+
+    attr_reader :units, :proc
```
Comment (zvkemp) [inline lib/tmx-parser/listener.rb:8]: Don't override keyword `proc`

File: Rakefile (hunk 2 above)
Comment (camertron) [inline Rakefile:5]: Actually I'm not sure... it's something I saw a bunch of other gems doing, so I copied and pasted :/

File: lib/tmx-parser/elements.rb (hunk 3 above)
Comment (camertron) [inline lib/tmx-parser/elements.rb]: Ah good call.

File: lib/tmx-parser/elements.rb (hunk 4 above)
Comment (camertron) [inline lib/tmx-parser/elements.rb:84]: They're distinct just so you can tell them apart by class name.

File: lib/tmx-parser/listener.rb (hunk 5 above)
Comment (camertron) [inline lib/tmx-parser/listener.rb:8]: I don't think proc is a keyword? The `Proc` constant is how you create new procs, eg `Proc.new`.

File: Gemfile (hunk 1 above)
Comment (camertron) [inline Gemfile]: Ok, can do. Out of curiosity, why is that better?

Comment (zvkemp) [inline Gemfile]: This is a bit old, but it's what I generally use for best practices on gemspec/bundle issues: http://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/

File: lib/tmx-parser/elements.rb (hunk 4 above)
Comment (zvkemp) [inline lib/tmx-parser/elements.rb:84]: But why do you need to? They have identical behavior, so from an OO point of view they are the same class.

File: lib/tmx-parser/listener.rb (hunk 5 above)
Comment (zvkemp) [inline lib/tmx-parser/listener.rb:8]: There's also `proc {|n| n + 1 }`

Comment (camertron) [inline lib/tmx-parser/listener.rb:8]: Huh... a quick Google search didn't reveal that, but it totally works in irb. Thanks for the #protip!

**Pull request merged**